src/azure-cli-core/azure/cli/core/extension/__init__.py

```diff
@@ -144,18 +144,11 @@ def get_metadata(self):
             metadata.update(azext_metadata)
 
         for dist_info_dirname in info_dirs:
-            parsed_dist_info_dir = WHEEL_INFO_RE(dist_info_dirname)
-            if not parsed_dist_info_dir:
-                continue
-
-            parsed_dist_info_dir = parsed_dist_info_dir.groupdict().get('name')
-            if os.path.split(parsed_dist_info_dir)[-1] == self.name.replace('-', '_'):
-                whl_metadata_filepath = os.path.join(dist_info_dirname, WHL_METADATA_FILENAME)
-                if os.path.isfile(whl_metadata_filepath):
-                    with open(whl_metadata_filepath) as f:
-                        metadata.update(json.loads(f.read()))
```

Comment:
By removing this logic, do new versions of Azure CLI still work with old extensions? If not, should we re-release all existing extensions?

Reply:
Yes, it works, even though the metadata structure for Python Package in CLI is changed.

Thanks to command part between the new structure and old stucture, I don't have to take care of the difference, because CLI only use name  property only among all the raw metadata defined in Python [Core Metadata Specification](https://packaging.python.org/specifications/core-metadata/). 

The new structure of metadata:
```
{'author': 'Matthew Booe',
 'author_email': 'mabooe@microsoft.com',
 'azext.isPreview': True,
 'azext.minCliCoreVersion': '2.2.0',
 'classifiers': ['Development Status :: 4 - Beta',
                 'Intended Audience :: Developers',
                 'Intended Audience :: System Administrators',
                 'Programming Language :: Python',
                 'Programming Language :: Python :: 2',
                 'Programming Language :: Python :: 2.7',
                 'Programming Language :: Python :: 3',
                 'Programming Language :: Python :: 3.5',
                 'Programming Language :: Python :: 3.6',
                 'Programming Language :: Python :: 3.7',
                 'Programming Language :: Python :: 3.8',
                 'License :: OSI Approved :: MIT License'],
 'home_page': 'https://github.com/Azure/azure-cli-extensions/tree/master/src/ai-examples',
 'license': 'MIT',
 'metadata_version': '2.0',
 'name': 'ai-examples',
 'platforms': ['UNKNOWN'],
 'summary': 'Add AI powered examples to help content.',
 'version': '0.2.0'}
```

The old structure:
```JSON
"azext.isPreview": true,
"azext.minCliCoreVersion": "2.0.81",
"classifiers": [
    "Development Status :: 4 - Beta",
    "Intended Audience :: Developers",
    "Intended Audience :: System Administrators",
    "Programming Language :: Python",
    "Programming Language :: Python :: 2",
    "Programming Language :: Python :: 2.7",
    "Programming Language :: Python :: 3",
    "Programming Language :: Python :: 3.5",
    "Programming Language :: Python :: 3.6",
    "Programming Language :: Python :: 3.7",
    "Programming Language :: Python :: 3.8",
    "License :: OSI Approved :: MIT License"
],
"extensions": {
    "python.details": {
        "contacts": [
            {
                "email": "mabooe@microsoft.com",
                "name": "Matthew Booe",
                "role": "author"
            }
        ],
        "document_names": {
            "description": "DESCRIPTION.rst"
        },
        "project_urls": {
            "Home": "https://github.com/Azure/azure-cli-extensions/tree/master/src/ai-examples"
        }
    }
},
"generator": "bdist_wheel (0.30.0)",
"license": "MIT",
"metadata_version": "2.0",
"name": "ai-examples",
"summary": "Add AI powered examples to help content.",
"version": "0.1.0"
```